m02_setup/01_first_steps.rst

```diff
@@ -0,0 +1,190 @@
+
+  git --version
+
+.. code-block:: bash
+  :caption: 3- Install build-essential, which includes make.
+
+  sudo apt update
+  sudo apt install build-essential
+
+.. code-block:: bash
+  :caption: 4- Verify make version
+
+  make --version
+
+::::
+
+All other software runs inside `Docker <https://www.docker.com/>`_ containers.
+For short, Docker is the new software platform for running A+ and software
+related to it. This course will discuss Docker later.
+
+Install Docker for Mac/Windows or docker-ce & docker-compose for Linux.
+
+- `Docker Community edition <https://docs.docker.com/engine/installation/>`_
+- `Docker Compose <https://docs.docker.com/compose/install/>`_
+
+You also need to install a Text editor (`VS Code <https://code.visualstudio.com/>`_  is recommended). You can find more information about the most convenient text editor to create *RST* documents :ref:`at the end of this module <text-editor-and-ides>`.
+
+Cloning this codebase with git
+------------------------------
+
+A good way to start is to clone the `Aplus course Template <https://github.com/apluslms/aplus-course-template>`_ with git and begin to work on it. In Aalto, you could use `version.aalto.fi repo <https://version.aalto.fi/gitlab/course/aplus-manual>`_. Please note that if you don't have access, then you can request access to the *Aalto IT services*. In that case, you probably should be part of :code:`apluslms-cs@aalto.fi` email list too.
+
+To clone, execute the following command:
+
+.. code-block:: sh
+
+    git clone git@version.aalto.fi:course/aplus-manual.git
+    cd aplus-manual
+
+In public, you could use `GitHub repo <https://github.com/apluslms/course-templates/>`_.
+
+.. code-block:: sh
+
+    git clone https://github.com/apluslms/course-templates.git
+    cd course-templates
+
+Next step, you need to get :code:`a-plus-rst-tools`.
+
+.. code-block:: sh
+
+    git submodule init
+    git submodule update
+
+Now all the course material is in RST format. Every time the RST part changes,
+you need to recompile it. You must do it also the first time. Open your terminal, go to the course directory and type the following command.
+
+.. code-block:: sh
+
+    ./docker-compile.sh
+
+This command runs Sphinx inside a Docker container. Sphinx reads all RST files.
+It produces compilation results into new directory ``_build``. That contains
+three subdirectories. ``doctrees`` has intermediate results of compilation.
+``html`` has the HTML versions of the RST documentation. ``yaml`` contains
+configuration files in `YAML format <https://en.wikipedia.org/wiki/YAML>`_
+and they are also meant for A+.
+
+For now, you can test the result by running A+ on your own computer:
+
+.. code-block:: sh
+
+    ./docker-up.sh
+
+Now A+ runs at http://localhost:8000. Open that address with your web browser.
+You must login by the maintenance login showing on the front page. The available
+users are `root`:`root` and `student`:`student`.  The default course is created
```

Comment:
We have added the user `teacher` that should be preferred for testing. Teachers don't have admin privileges in the production server. `Root` has them in the container.